.github/workflows/build-distributables.yml

```diff
@@ -29,26 +33,145 @@ jobs:
     - uses: conda-incubator/setup-miniconda@v3
       with:
         auto-update-conda: true
-        python-version: ${{ matrix.config.py }}
+        python-version: ${{ env.PYTHON_VERSION }}
+        miniforge-version: latest
+        activate-environment: builder
 
     - name: Build conda-packed (all platforms)
-      shell: bash -el {0}
       run: |
+        conda install -y versioningit nodejs
         conda activate base
         ./extra/build_conda_packed.sh
-        ls artifacts/*
-        {
-          echo 'DISTRIBUTABLES<<EOF'
-          ls artifacts/*
-          echo EOF
-        } >> "$GITHUB_ENV"
+
+    - name: Create MacOS App
+      if: startsWith(matrix.config.os, 'macos')
+      env:
+          MACOS_CERTIFICATE: ${{ secrets.MACOS_CERTIFICATE_ISA }}
+          MACOS_CERTIFICATE_PWD: ${{ secrets.MACOS_CERTIFICATE_ISA_PWD }}
+          KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+          NOTARIZATION_USERNAME: ${{ secrets.MACOS_NOTARIZATION_USERNAME }}
+          NOTARIZATION_PASSWORD: ${{ secrets.MACOS_NOTARIZATION_PASSWORD }}
+          DEVELOPER_IDENTITY: "Developer ID Application: The International Scattering Alliance (8CX8K63BQM)"
+      run: |
+        echo $MACOS_CERTIFICATE | base64 --decode > certificate.p12
+        security create-keychain -p $KEYCHAIN_PASSWORD build.keychain
+        security default-keychain -s build.keychain
+        security unlock-keychain -p $KEYCHAIN_PASSWORD build.keychain
+        security import certificate.p12 -k build.keychain -P $MACOS_CERTIFICATE_PWD -T /usr/bin/codesign
+        security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k $KEYCHAIN_PASSWORD build.keychain
+        security find-identity -p codesigning
+
+        echo "Creating MacOS App"
+        # This assumes that conda-packed enviroment exists at "conda_packed"
+
+        VERSION=$(versioningit)
+        PRODUCT_NAME="$PACKAGE_NAME-$VERSION"
```

Comment:
Can we put PRODUCT_NAME, PRODUCT_VERSION and COPYRIGHT in a common environment so they are shared across deployment platforms?